src/otx/algo/utils/xai_utils.py

```diff
@@ -18,19 +20,22 @@
 def get_processed_saliency_maps(
     explain_hook: BaseRecordingForwardHook,
     explain_config: ExplainConfig,
-    predictions: list | None,
+    predictions: list[Any] | list[OTXBatchPredEntityWithXAI | InstanceSegBatchPredEntityWithXAI] | None,
     work_dir: Path | None,
 ) -> list:
     """Implement saliency map filtering and post-processing."""
     # Optimize for memory <- TODO(negvet)
-    raw_saliency_maps = explain_hook.records
+    raw_saliency_maps: list = []
+
+    if predictions is not None:
+        raw_saliency_maps = predictions[0].saliency_maps
 
     if predictions is not None and isinstance(explain_hook, MaskRCNNRecordingForwardHook):
         # TODO: It is a temporary workaround. This function will be removed after we # noqa: TD003, TD002
         # refactor XAI logics into `OTXModel.forward_explain()`.
 
         # Mask-RCNN case, receive saliency maps from predictions
```

Comment:
```suggestion
    if predictions is not None:
        raw_saliency_maps = predictions[0].saliency_maps

        if isinstance(explain_hook, MaskRCNNRecordingForwardHook):
            # TODO: It is a temporary workaround. This function will be removed after we # noqa: TD003, TD002
            # refactor XAI logics into `OTXModel.forward_explain()`.

            # Mask-RCNN case, receive saliency maps from predictions
```

Reply:
To me this logic is unclear, end even more, has no sense in regards with the upcoming changes, because we should not have any exceptional case for the iseg results.